plugins/portmap/README.md

```diff
@@ -0,0 +1,103 @@
+## Port-mapping plugin
+
+This plugin will forward traffic from one or more ports on the host to the
+container. It expects to be run as a chained plugin.
+
+## Usage
+You should use this plugin as part of a network configuration list. It accepts
+the following configuration options:
+
+* `noSnat` - boolean, defaul false. If true, do not set up the SNAT chains
+
+The plugin expects to receive the actual list of port mappings via the 
+[`portMappings` capability argument](CONVENTIONS.md#plugin-specific-fields)
+
+So a typical config may look like,
+
+```json
+{
+        "cniVersion": "0.3.0",
+        "name": "mynet",
+        "plugins": [
+                {
+                        "type": "ptp",
+                        "ipMasq": true,
+                        "ipam": {
+                                "type": "host-local",
+                                "subnet": "172.16.30.0/24",
+                                "routes": [
+                                        {
+                                                "dst": "0.0.0.0/0"
+                                        }
+                                ]
+                        }
+                },
+                {
+                        "type": "portmap",
+						"capabilities": {"portMappings": true}
+                }
+        ]
+}
+```
+
+
+## Rule structure
+The plugin sets up two sequences of chains and rules - one "primary" DNAT
+sequence to rewrite the destination, and one additional SNAT sequence that
+rewrites the source address for packets from localhost. The sequence is somewhat
+complex to minimize the number of rules non-forwarded packets must traverse.
+
+
+### DNAT
+The DNAT rule rewrites the destination port and address of new connections.
+There is a top-level chain, `CNI-HOSTPORT-DNAT` which is always created and
+never deleted. Each plugin execution creates an additional chain for ease
+of cleanup. So, if a single container exists on IP 172.16.30.2 with ports 
+8080 and 8043 on the host forwarded to ports 80 and 443 in the container, the 
+rules look like this:
+
+`PREROUTING`, `OUTPUT` chains:
```

Comment:
Something we discovered in Kubernetes - this expression is looser than we really want.  This will match any address in the `local` routing table, whereas I think we only want addresses that are actually bound to interfaces.  The delta here is things like load-balancer VIPs on GCE.

The solutions that is pending is to loop over all interfaces and expand them into a set of exact-IP matches.  It is sort of awful because, at least in theory, this can change, so it needs to be refreshed periodically. I don't know how else to express this condition, but I am open to ideas.

An alternative would be to document this as "must be the last rule in the chain" but that feels very brittle in the face of collaboration users of iptables.